File: src/main/java/catwalks/compat/jei/DecorationSplitRecipeHandler.java
package catwalks.compat.jei;

import catwalks.item.crafting.RecipeDecorationSplit;
import mezz.jei.api.IGuiHelper;
import mezz.jei.api.ingredients.IIngredients;
import mezz.jei.api.recipe.BlankRecipeWrapper;
import mezz.jei.api.recipe.IRecipeHandler;
import mezz.jei.api.recipe.IRecipeWrapper;
import mezz.jei.api.recipe.VanillaRecipeCategoryUid;
import mezz.jei.api.recipe.wrapper.ICraftingRecipeWrapper;
import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.client.resources.I18n;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;

import javax.annotation.Nonnull;
import java.util.ArrayList;
import java.util.List;

public class DecorationSplitRecipeHandler implements IRecipeHandler<RecipeDecorationSplit> {

	IGuiHelper helper;
	
	public DecorationSplitRecipeHandler(IGuiHelper helper) {
		this.helper = helper;
	}
	
	@Nonnull
	@Override
	public Class getRecipeClass() {
		return RecipeDecorationSplit.class;
	}

	@Nonnull
	@Override
	public String getRecipeCategoryUid() {
		return VanillaRecipeCategoryUid.CRAFTING;
	}
	
	@Nonnull
	@Override
	public String getRecipeCategoryUid(@Nonnull RecipeDecorationSplit recipe) {
		return VanillaRecipeCategoryUid.CRAFTING;
	}
	
	@Nonnull
	@Override
	public IRecipeWrapper getRecipeWrapper(@Nonnull RecipeDecorationSplit recipe) {
		return new Wrapper(helper, recipe);
	}

	@Override
	public boolean isRecipeValid(@Nonnull RecipeDecorationSplit recipe) {
		return true;
	}
	
	public static class Wrapper extends BlankRecipeWrapper implements ICraftingRecipeWrapper {

		List inputs, outputs;
		
		@SuppressWarnings("unchecked")
		public Wrapper(IGuiHelper guiHelper, RecipeDecorationSplit recipe) {
			Item item = recipe.getTargetItem();
			inputs =  new ArrayList<>();
			outputs = new ArrayList<>();
			
			inputs.add(new ItemStack(item));
			outputs.add(new ItemStack(item, 1, item.getMaxDamage() - (item.getMaxDamage()/2)));
		}
		
		@Override
		public void getIngredients(IIngredients ingredients) {
			ingredients.setInputs(ItemStack.class, inputs);
			ingredients.setOutputs(ItemStack.class, outputs);
		}
		
		@Nonnull
		@Override
		public List getInputs() {
			return inputs;
		}

		@Nonnull
		@Override
		public List<ItemStack> getOutputs() {
			return outputs;
		}
		
		@Override
		public void drawInfo(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight, int mouseX, int mouseY) {
			super.drawInfo(minecraft, recipeWidth, recipeHeight, mouseX, mouseY);

			GlStateManager.pushMatrix();
			
			String str = I18n.format("gui.jei.crafting.decorationSplit");
			int width = minecraft.fontRendererObj.getStringWidth(str);
			
			minecraft.fontRendererObj.drawString(str, 103-(width/2), 41, 4210752);
			
			GlStateManager.popMatrix();
		}
		
	}

}

File: src/main/java/catwalks/compat/jei/DecorationSplitRecipeHandler.kt
package catwalks.compat.jei

import catwalks.item.crafting.RecipeDecorationSplit
import mezz.jei.api.IGuiHelper
import mezz.jei.api.ingredients.IIngredients
import mezz.jei.api.recipe.BlankRecipeWrapper
import mezz.jei.api.recipe.IRecipeHandler
import mezz.jei.api.recipe.IRecipeWrapper
import mezz.jei.api.recipe.VanillaRecipeCategoryUid
import mezz.jei.api.recipe.wrapper.ICraftingRecipeWrapper
import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.GlStateManager
import net.minecraft.client.resources.I18n
import net.minecraft.item.ItemStack

class DecorationSplitRecipeHandler(internal var helper:

                                   IGuiHelper) : IRecipeHandler<RecipeDecorationSplit> {

    override fun getRecipeClass(): Class<RecipeDecorationSplit> {
        return RecipeDecorationSplit::class.java
    }

    override fun getRecipeCategoryUid(): String {
        return VanillaRecipeCategoryUid.CRAFTING
    }

    override fun getRecipeCategoryUid(recipe: RecipeDecorationSplit): String {
        return VanillaRecipeCategoryUid.CRAFTING
    }

    override fun getRecipeWrapper(recipe: RecipeDecorationSplit): IRecipeWrapper {
        return Wrapper(helper, recipe)
    }

    override fun isRecipeValid(recipe: RecipeDecorationSplit): Boolean {
        return true
    }

    class Wrapper
    @SuppressWarnings("unchecked")
    constructor(guiHelper: IGuiHelper, recipe: RecipeDecorationSplit) : BlankRecipeWrapper(), ICraftingRecipeWrapper {

        internal val inputs = mutableListOf<ItemStack>()
        internal val outputs = mutableListOf<ItemStack>()

        init {
            val item = recipe.targetItem

            inputs.add(ItemStack(item))
            outputs.add(ItemStack(item, 1, item.maxDamage - item.maxDamage / 2))
        }

        override fun getIngredients(ingredients: IIngredients) {
            ingredients.setInputs<ItemStack>(ItemStack::class.java, inputs)
            ingredients.setOutputs<ItemStack>(ItemStack::class.java, outputs)
        }

        override fun getInputs(): List<*> {
            return inputs
        }

        override fun getOutputs(): List<ItemStack> {
            return outputs
        }

        override fun drawInfo(minecraft: Minecraft, recipeWidth: Int, recipeHeight: Int, mouseX: Int, mouseY: Int) {
            super.drawInfo(minecraft, recipeWidth, recipeHeight, mouseX, mouseY)

            GlStateManager.pushMatrix()

            val str = I18n.format("gui.jei.crafting.decorationSplit")
            val width = minecraft.fontRendererObj.getStringWidth(str)

            minecraft.fontRendererObj.drawString(str, 103 - width / 2, 41, 4210752)

            GlStateManager.popMatrix()
        }

    }

}
